text/0087-graphql-errors.md

```diff
@@ -0,0 +1,260 @@
+
+The `Request` and `Response` bodies could contain PII.
+
+Request body example (query language only):
+
+```
+query { 
+  viewer { 
+    login
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "data": {
+    "viewer": {
+      "login": "marandaneto"
+    }
+  }
+}
+```
+
+A Request example with an error (query language only):
+
+```
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    logi
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "errors": [
+    {
+      "path": [
+        "query",
+        "viewer",
+        "logi"
+      ],
+      "extensions": {
+        "code": "undefinedField",
+        "typeName": "User",
+        "fieldName": "logi"
+      },
+      "locations": [
+        {
+          "line": 4,
+          "column": 5
+        }
+      ],
+      "message": "Field 'logi' doesn't exist on type 'User'"
+    }
+  ]
+}
+```
+
+Using the `locations` and the request body (query language only), we can highlight the error in the request body.
+
+```
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    -> logi
+  }
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
+
+The Request body can also contain `variables`.
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {
+    "login": "marandaneto"
+  }
+}
+```
+
+Because of that, the `Request` and `Response` body should be sent to Sentry.
+
+# Supporting Data
+
+See [JIRA issues](https://www.notion.so/sentry/GraphQL-proposal-d6e5846f30434770903cf3af20bc2568?pvs=4#fd0b946340d947eb9bbaa320846ccd6a), not disclosing here because they could contain PII.
+
+Related issues and discussions:
+
+[GraphQL Support](https://github.com/getsentry/sentry/discussions/38913)
+
+[Support for GraphQL errors](https://github.com/getsentry/sentry/issues/33723)
+
+[Send request body for http client integrations and similar](https://github.com/getsentry/team-mobile/issues/41)
+
+# Proposal (Option 1)
+
+The proposal is adding a `data` field in the [Response](https://develop.sentry.dev/sdk/event-payloads/types/#responsecontext) interface.
+
+By doing this, we can keep the `Request` interface as it is, we can copy the `data` scrubbing rules from the `Request` interface.
+
+```json
+{
+  "contexts": {
+    "response": {
+      "type": "response",
+      "status_code": 500,
+      "body_size": 1000,
+      "data": "...",
+    }
+  }
+}
+```
+
+* `data`: Can be given as string or structural data of any format..
+
+The `Response` interface keeps arbitrary fields, it's back compatible with the current implementation.
+
+## Must have
+
+The fields `Request#data` and `Response#data` could contain PII and they should run data scrubbing agressively.
```

Comment:
We should consider allowing people to opt out of scrubbing. Often, the actual payload is the information needed to reproduce bugs.

Reply:
Which project config? Are you talking about server-side scrubbing?

Reply:
Yes, eg https://sentry-sdks.sentry.io/settings/projects/sentry-cocoa/security-and-privacy/
Scrubbing will be server-side, so if there's an option to enable/disable, it should be there.
People can also opt-in for SDK scrubbing using `beforeSend`.